### PR TITLE
IBP-4858 Scale type restriction for Attributes and Passport variable types.

### PR DESCRIPTION
**Pull request by cuenyad**
Please, review.

### CODE REVIEW

File: src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/VariableValidator.java [hunk 1]
```diff
@@ -515,9 +535,9 @@ void variableShouldBeEditable(final VariableDetails variable, final Errors error
 			final boolean propertyEqual = Objects.equals(propertyId, oldVariable.getProperty().getId());
 			final boolean methodEqual = Objects.equals(methodId, oldVariable.getMethod().getId());
 			final boolean scaleEqual = Objects.equals(scaleId, oldVariable.getScale().getId());
-			final boolean minValuesEqual = editableVariable ? true : StringUtil.areBothEmptyOrEqual(variable.getExpectedRange().getMin(), oldVariable.getMinValue());
-			final boolean maxValuesEqual = editableVariable ? true : StringUtil.areBothEmptyOrEqual(variable.getExpectedRange().getMax(), oldVariable.getMaxValue());
-			final boolean aliasEqual = editableVariable ? true : StringUtil.areBothEmptyOrEqual(variable.getAlias(), oldVariable.getAlias());
+			final boolean minValuesEqual = editableVariable ? editableVariable : StringUtil.areBothEmptyOrEqual(variable.getExpectedRange().getMin(), oldVariable.getMinValue());
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/VariableValidator.java:538]: why this change is required for the file data type limitation? 
or was it a bug? if it was a bug, then should be included to be tested
@cuenyad

File: src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/VariableValidator.java [hunk 2]
```diff
@@ -389,6 +397,18 @@ private boolean scaleDataTypeValidationProcessor(final VariableDetails variable,
 				}
 			}
 
+			final boolean isFileType = Objects.equals(scale.getDataType(), DataType.FILE_VARIABLE);
+			// Temporally restriction for Germplasm Attributes and Passport Variable Types.
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/VariableValidator.java:401]: Fix typo: Temporally to temporal
Add reference to IBP-4798, this restriction should be removed when implementing this ticket.

**Review by clarysabel — CHANGES_REQUESTED** [review]
Small changes. 
I could see some changes that are not exclusively from the restriction implementation. 
If that's the case, please edit the testing instructions and add regression testing for the scenarios that you have modified

File: src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/VariableValidator.java (hunk 1 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/VariableValidator.java:538]: Yes, it's gonna be tested. 
When I added the restricction made a SonaLint cleanUp.

File: src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/VariableValidator.java (hunk 2 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/VariableValidator.java]: done

**Comment by clarysabel** [comment]
Not required because the File data type was deleted after implementing IBP-4815
